### PR TITLE
Add initial analytics/sensors module.

### PR DESCRIPTION
**Pull request by ddebrunner**
Still needs some javadoc and testing for the deadband filter with maximum suppression time.

### CODE REVIEW

File: analytics/sensors/src/main/java/quarks/analytics/sensors/Filters.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+     * @param <V> Value type for the deadband function.
+     * 
+     * @param stream Stream containing readings.
+     * @param value Function to obtain the tuple's value passed to the deadband function.
+     * @param deadband Function that defines the deadband.
+     * @param maximumSuppression Maximum amount of time to suppress values that are in the deadband.
+     * @param unit Unit for {@code maximumSuppression}.
+     * @return Filtered stream.
+     */
+    public static <T, V> TStream<T> deadband(TStream<T> stream, Function<T, V> value, Predicate<V> deadband,
```
Comment (dlaboss) [inline analytics/sensors/src/main/java/quarks/analytics/sensors/Filters.java]: Might just be me but I'm finding things confusing because the name "deadband" is used multiple times but with opposite semantics.  

i.e., `Predicate deadband` is testing if the value is IN the deadband - i.e., uninteresting values, whereas `static deadband()`, something with the exact same name, is selecting tuples OUT of the deadband - i.e., interesting values.

`DeadbandFilter` also has the OUT-of-deadband semantics and also takes the `Predicate deadband` which has the IN-deadband semantics.   Also, I think its not helping that `TStream.filter()` doesn't change the sense of what its predicate selects, "filter()" doesn't "filter out" what the predicate selects, but `DeadbandFilter` does.

Perhaps changing to `Predicate inDeadband` will at least help differentiate it & clarify what its semantics are?

Comment (ddebrunner) [inline analytics/sensors/src/main/java/quarks/analytics/sensors/Filters.java]: yeah, I struggled with that, the filter predicate being inverted, but it seemed clearer to have the predicate be "positive" and define what was the deadband is, rather than what it is not.

I think the name was based upon some disucssions with someone who implemented this functionality for sensors in the oil/gas industry.

I'll change the predicate name to `inPredicate` and think about the `DeadbandFilter` name, though I think that's not part of the api.

Comment (ddebrunner) [inline analytics/sensors/src/main/java/quarks/analytics/sensors/Filters.java]: One alternative is to switch to:

`public static <T, V> TStream<T> outOBand(TStream<T> stream, Function<T, V> value, Predicate<V> inBand)`

Comment (dlaboss) [inline analytics/sensors/src/main/java/quarks/analytics/sensors/Filters.java]: Yeah, that sort of thing occurred to me too.  I like that In/Out distinction.  But  "outOfBand()" is misleading as these streams also include in-band values under the appropriate circumstances.  In that regard, "deadband()" seems better - but still use "inBand" for the predicate name.

I hadn't noticed that `DeadbandFilter` wasn't public.  Regardless, why not just drop the `Filter` and call it `Deadband` -- it's a predicate that selects/has the same semantics as `deadband()`.
